libraries/chain/util/smt_token.cpp

```diff
@@ -428,6 +443,51 @@ bool schedule_founder_payout( database& db, const asset_symbol_type& a )
    return action_scheduled;
 }
 
+fc::optional< share_type > steem_units_hard_cap( database& db, const asset_symbol_type& a )
+{
+   const auto& idx = db.get_index< smt_ico_tier_index, by_symbol_steem_units_cap >();
+
+   const auto range = idx.equal_range( a );
+
+   auto itr = range.second;
+   while ( itr != range.first )
+   {
+      --itr;
+      return itr->steem_units_cap;
+   }
+
+   return {};
+}
+
+void remove_objects( database& db, const asset_symbol_type& symbol )
+{
+   db.remove( db.get< smt_ico_object, by_symbol >( symbol ) );
+
+   const auto& ico_tier_idx = db.get_index< smt_ico_tier_index, by_symbol_steem_units_cap >();
+   auto itr = ico_tier_idx.lower_bound( symbol );
+   while( itr != ico_tier_idx.end() && itr->symbol == symbol )
+   {
+      const auto& current = *itr;
+      ++itr;
+      db.remove( current );
+   }
+}
+
+std::size_t tier_size( database& db, const asset_symbol_type& symbol )
```

Comment:
This implementation is fine for a small ico tier cap, but we may want to make a note to track this in the ico object if we increase the number of ICO tiers allowed.